catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergConfigurer.java

```diff
@@ -339,17 +245,77 @@ IcebergTableConfig icebergConfigPerTable(
       }
     }
 
-    StorageLocations storageLocations =
+    StorageLocations locations =
         storageLocations(StorageUri.of(warehouseLocation), writeable, readOnly);
 
     Predicate<AccessDelegation> accessDelegationPredicate = accessDelegationPredicate(dataAccess);
 
     Map<String, String> config = new HashMap<>();
+    objectIO.icebergTableConfig(
+        locations,
+        config::put,
+        storageLocations -> {
+          // Handling for S3 signing is very much integrated w/ request-URI/context, so that
+          // functionality stays here and is not handled in S3ObjectIO.
+
+          if (!accessDelegationPredicate.test(REMOTE_SIGNING)) {
+            return false;
+          }
+          if (!Stream.concat(
+                  storageLocations.writeableLocations().stream(),
+                  storageLocations.readonlyLocations().stream())
+              .map(StorageUri::scheme)
+              .allMatch(S3Utils::isS3scheme)) {
+            return false;
+          }
+
+          String normalizedWarehouseLocation = normalizeS3Scheme(warehouseLocation);
+
+          // Must use both 's3.signer.uri' and 's3.signer.endpoint', because Iceberg before 1.5.0
+          // does
+          // not handle full URIs passed via 's3.signer.endpoint'. This was changed via
+          // https://github.com/apache/iceberg/pull/8976/files#diff-1f7498b6989fffc169f7791292ed2ccb35b305f6a547fd832f6724057c8aca8bR213-R216,
+          // first released in Iceberg 1.5.0. It's unclear how other language implementations deal
+          // with this.
+          config.put(S3_SIGNER_URI, uriInfo.icebergBaseURI().toString());
```

Comment:
Hmmm, changing S3 `config` inside a "predicate" looks pretty obscure to me... Especially considering that some `ObjectIO` implementations do not call it at all.

I'd also move the setting of `S3_REMOTE_SIGNING_ENABLED` here.

Also, the predicate checks `locations`, which we already know... why add a round trip through a `Predicate`?

Reply:
Maybe add a separate `ObjectIO.configureRemoteSigning()` method?

Reply:
All the URI stuff - that definitely doesn't belong to ObjectIO. Too complex to get it into ObjectIO

Reply:
I'm +1 to keep it out of `ObjectIO`. 

Still, WDYT about removing the `Predicate` parameter and instead doing `if(isS3(locations)) {...}` here (including the setting of `S3_REMOTE_SIGNING_ENABLED`)?

Reply:
Not really possible - see the comment I've added